src/Proto3/Wire/Decode.hs

```diff
@@ -441,6 +458,26 @@ sfixed64 = runGetFixed64 getInt64le
 at :: Parser RawField a -> FieldNumber -> Parser RawMessage a
 at parser fn = Parser $ runParser parser . fromMaybe mempty . M.lookup fn
 
+-- | Try to parse different field numbers with their respective parsers.
+-- This is used to express alternative between possible fields
+--
+-- If no field number are present for the oneof, then the default
+-- behavior of the first parser is used.
+--
+-- TODO: contrary to the protobuf spec, in the case of multiple fields number
```

Comment:
Is this comment still necessary after your change to use an association list instead of a `Map`?

Reply:
provided you're happy going down this path and that doesn't raise compatibility questions, and the rest of proto3-suite doesn't depends on these anymore, then yes many parts shouldn't be needed anymore including this oneof function.